Clarify plugin_tmpdir for all supported Vault versions (#2771)
hashicorp/web-unified-docs: content/vault/v1.19.x/content/docs/configuration/index.mdx

  This only needs to be set if the file permissions check is enabled via the environment variable
  `VAULT_ENABLE_FILE_PERMISSIONS_CHECK`.

> 
> - `plugin_tmpdir` `(string: "")` - A directory Vault uses for temporary files
>   when starting [containerized external plugins](/vault/docs/plugins/containerized-plugins).
>   Has no effect on non-containerized plugins. Set `plugin_tmpdir` to a path accessible to both Vault and
>   the plugin container when the default OS temporary directory is not shared between them. 
>   Set `VAULT_PLUGIN_TMPDIR` to configure the directory with an environment variable.

Reviewer: ```suggestion
- `plugin_tmpdir` `(string: "")` - A directory Vault uses for temporary files
  when starting [containerized external plugins](/vault/docs/plugins/containerized-plugins).
  You must set `plugin_tmpdir` to a path that Vault and the plugin container can
  both access when they do not share the default OS temp directory. Vault does
  not use `plugin_tmpdir` with non-containerized plugins. Set
  `VAULT_PLUGIN_TMPDIR` to configure the directory with an environment variable.
```
Style correction: use simple sentences whenever possible, write in active voice
